### PR TITLE
ref(ts): convert IssueListFilters

### PR DESCRIPTION
**Pull request by davidenwang**
convert to tsx

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/filters.tsx [hunk 1]
```diff
@@ -2,52 +2,49 @@ import PropTypes from 'prop-types';
...
-    onSavedSearchDelete: PropTypes.func.isRequired,
-    tagValueLoader: PropTypes.func.isRequired,
-    tags: PropTypes.object.isRequired,
-  };
+const defaultProps = {
+  sort: '',
+  query: null as string | null,
+  onSortChange: (() => {}) as (sort: string) => void,
+  onSearch: (() => {}) as (query: string) => void,
+  onSidebarToggle: (() => {}) as (event: React.MouseEvent) => void,
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/views/issueList/filters.tsx:21]: Would it be more legible to split this up? 

```
type DefaultProps = {
  ...
};

type Props = {
  ...
} & DefaultProps;

const defaultProps: DefaultProps = {
  ...
};
```

**Review by leedongwei — COMMENTED** [review]
Hi David.

File: src/sentry/static/sentry/app/views/issueList/filters.tsx (hunk 1 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/issueList/filters.tsx:21]: Do we actually need the function default props? The component is used in ~one place and should be provided

Comment (davidenwang) [reply src/sentry/static/sentry/app/views/issueList/filters.tsx:21]: ohh shoot just noticed that. In that case I'll just remove the default props and adjust accordingly

**Review by markstory — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

File: src/sentry/static/sentry/app/views/issueList/filters.tsx [hunk 2]
```diff
@@ -2,52 +2,44 @@ import PropTypes from 'prop-types';
...
+  isSearchDisabled: boolean;
+  queryCount: number;
+  queryMaxCount: number;
 
+  onSortChange: (sort: string) => void;
+  onSearch: (query: string) => void;
+  onSidebarToggle: (event: React.MouseEvent) => void;
+  onSavedSearchSelect: (search: SavedSearch) => void;
+  onSavedSearchDelete: (search: SavedSearch) => void;
+  tagValueLoader: React.ComponentProps<typeof IssueListSearchBar>['tagValueLoader'];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/issueList/filters.tsx:33]: maybe you could create:  `type IssueListSearchBarProps = React.ComponentProps<typeof IssueListSearchBar>`

and reuse it?

File: src/sentry/static/sentry/app/views/issueList/filters.tsx [hunk 3]
```diff
@@ -2,52 +2,44 @@ import PropTypes from 'prop-types';
...
-    onSearch: PropTypes.func,
-    onSidebarToggle: PropTypes.func,
-    onSavedSearchSelect: PropTypes.func.isRequired,
-    onSavedSearchDelete: PropTypes.func.isRequired,
-    tagValueLoader: PropTypes.func.isRequired,
-    tags: PropTypes.object.isRequired,
-  };
+type Props = {
+  organization: Organization;
+
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/issueList/filters.tsx:18]:
```suggestion
```

File: src/sentry/static/sentry/app/views/issueList/filters.tsx [hunk 4]
```diff
@@ -2,52 +2,44 @@ import PropTypes from 'prop-types';
...
+  savedSearchList: SavedSearch[];
+  savedSearch: SavedSearch;
+
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/issueList/filters.tsx:21]:
```suggestion
```

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions, but it looks good 🙃

**Pull request merged**